io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -243,64 +507,117 @@ impl Share for Lvol {
     fn bdev_uri_original(&self) -> Option<String> {
         None
     }
-}
+}
+
+/// It will give the Common Interface for Volume and Snapshots
+#[async_trait(?Send)]
+pub trait LogicalVolume {
+    /// returns the name of the Logical Volume
+    fn name(&self) -> String;
+
+    /// returns the UUID of the Logical Volume
+    fn uuid(&self) -> String;
+
+    /// return lvs for the Logical Volume
+    fn lvs(&self) -> Lvs;
+
+    /// returns the pool name of the Logical Volume
+    fn pool_name(&self) -> String;
+
+    /// returns the pool uuid of the Logical Volume
+    fn pool_uuid(&self) -> String;
+
+    /// returns a boolean indicating if the Logical Volume is thin provisioned
+    fn is_thin(&self) -> bool;
+
+    /// returns a boolean indicating if the Logical Volume is read-only
+    fn is_read_only(&self) -> bool;
+
+    /// return the size of the Logical Volume in bytes
+    fn size(&self) -> u64;
+
+    /// Returns Lvol disk space usage.
+    fn usage(&self) -> LvolSpaceUsage;
+}
+
+/// It will give Interface Specific to Volume
+#[async_trait(? Send)]
+pub trait VolumeOps: LogicalVolume {
+    // wipe the first 8MB if unmap is not supported on failure the operation
+    // needs to be repeated
+    async fn wipe_super(&self) -> Result<(), Error>;
+
+    /// returns a boolean indicating if the lvol is a snapshot
+    fn is_snapshot(&self) -> bool;
+
+    /// get/read a property from this lvol from disk
+    async fn get(&self, prop: PropName) -> Result<PropValue, Error>;
```

Comment:
```suggestion
    async fn read_property(&self, prop: PropName) -> Result<PropValue, Error>;
```
?

Reply:
It is handled in the current changes. The older changes with Box based dyn trait is removed now.